curriculum/challenges/spanish/02-javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords.spanish.md

```diff
@@ -7,10 +7,10 @@ localeTitle: Comparar los alcances de la var y dejar Palabras clave
+<section id="description"> Cuando se declara una variable con la palabra clave <code>var</code> , se declara globalmente o localmente dependiendo de si esta se declara dentro de una función. La palabra clave <code>let</code> se comporta de una manera similar, pero con algunas características adicionales. Cuando se declara una variable con la palabra clave <code>let</code> dentro de un bloque, una declaración o una expresión, su alcance se limita a ese bloque, declaración o expresión. Por ejemplo: <blockquote> var numArray = []; <br> for(var i = 0; i &lt;3; i ++) { <br> numArray.push(i); <br> } <br> console.log(numArray); <br> // devuelve [0, 1, 2] <br> console.log(i); <br> // devuelve 3 </blockquote> Con la palabra clave <code>var</code>, <code>i</code> es declarada a nivel global. Entonces, cuando se ejecuta <code>i++</code> , se actualiza la variable global. Este código es similar al siguiente: <blockquote> var numArray = []; <br> var i; <br> for(i = 0; i &lt;3; i ++) { <br> numArray.push (i); <br> } <br> console.log (numArray); <br> // devuelve [0, 1, 2] <br> console.log(i); <br> // devuelve 3 </blockquote> Este comportamiento causaría problemas si se tuviera que crear una función y almacenarla para su uso posterior dentro de un bucle "for" que usa la variable <code>i</code> . Esto se debe a que la función almacenada siempre se referirá al valor de la variable global actualizada <code>i</code> . <blockquote> var printNumTwo; <br> for(var i = 0; i &lt;3; i ++) { <br> if(i === 2) { <br> printNumTwo = función () { <br> return i; <br> }; <br> } <br> } <br> console.log(printNumTwo ()); <br> // devuelve 3 </blockquote> Como se puede ver, <code>printNumTwo()</code> imprime 3 y no 2. Esto es porque el valor asignado a <code>i</code> se actualiza y <code>printNumTwo()</code> devuelve el valor global <code>i</code> y no el valor que<code>i</code> tenía cuando la función fue creada en el bucle. La palabra clave <code>let</code> no sigue este comportamiento: <blockquote> &#39;use strict&#39;; <br> let printNumTwo; <br> for(let i = 0; i &lt;3; i ++) { <br> if(i === 2) { <br> printNumTwo = function() { <br> return i; <br> }; <br> } <br> } <br> console.log(printNumTwo ()); <br> // devuelve 2 <br> console.log(i); <br> // devuelve &quot;i undefined&quot; </blockquote> <code>i</code> no está definido porque no se declaró en el ámbito global. Solo se declara dentro de la sentencia for loop. <code>printNumTwo()</code> devolvió el valor correcto porque se crearon tres variables <code>i</code> diferentes, con valores únicos (0, 1 y 2) mediante la palabra clave <code>let</code> dentro de la instrucción de bucle. </section>
```

Comment:
```suggestion
<section id="description"> Cuando se declara una variable con la palabra clave <code>var</code> , se declara globalmente o localmente dependiendo de si esta se declara dentro de una función. La palabra clave <code>let</code> se comporta de una manera similar, pero con algunas características adicionales. Cuando se declara una variable con la palabra clave <code>let</code> dentro de un bloque, una declaración o una expresión, su alcance se limita a ese bloque, declaración o expresión. Por ejemplo: <blockquote> var numArray = []; <br> for(var i = 0; i &lt;3; i ++) { <br> numArray.push(i); <br> } <br> console.log(numArray); <br> // devuelve [0, 1, 2] <br> console.log(i); <br> // devuelve 3 </blockquote> Con la palabra clave <code>var</code>, <code>i</code> es declarada a nivel global. Entonces, cuando se ejecuta <code>i++</code> , se actualiza la variable global. Este código es similar al siguiente: <blockquote> var numArray = []; <br> var i; <br> for(i = 0; i &lt;3; i ++) { <br> numArray.push (i); <br> } <br> console.log (numArray); <br> // devuelve [0, 1, 2] <br> console.log(i); <br> // devuelve 3 </blockquote> Este comportamiento causaría problemas si se tuviera que crear una función y almacenarla para su uso posterior dentro de un bucle "for" que usa la variable <code>i</code> . Esto se debe a que la función almacenada siempre se referirá al valor de la variable global actualizada <code>i</code> . <blockquote> var printNumTwo; <br> for(var i = 0; i &lt;3; i ++) { <br> if(i === 2) { <br> printNumTwo = function () { <br> return i; <br> }; <br> } <br> } <br> console.log(printNumTwo ()); <br> // devuelve 3 </blockquote> Como se puede ver, <code>printNumTwo()</code> imprime 3 y no 2. Esto es porque el valor asignado a <code>i</code> se actualiza y <code>printNumTwo()</code> devuelve el valor global <code>i</code> y no el valor que<code>i</code> tenía cuando la función fue creada en el bucle. La palabra clave <code>let</code> no sigue este comportamiento: <blockquote> &#39;use strict&#39;; <br> let printNumTwo; <br> for(let i = 0; i &lt;3; i ++) { <br> if(i === 2) { <br> printNumTwo = function() { <br> return i; <br> }; <br> } <br> } <br> console.log(printNumTwo ()); <br> // devuelve 2 <br> console.log(i); <br> // devuelve &quot;i undefined&quot; </blockquote> <code>i</code> no está definido porque no se declaró en el ámbito global. Solo se declara dentro de la sentencia for loop. <code>printNumTwo()</code> devolvió el valor correcto porque se crearon tres variables <code>i</code> diferentes, con valores únicos (0, 1 y 2) mediante la palabra clave <code>let</code> dentro de la instrucción de bucle. </section>
```
missed one funcion, otherwise LGTM.